src/api/OnmsServer.ts

```diff
@@ -34,27 +126,20 @@ export class OnmsServer {
    * ```javascript
    * const server = new OnmsServer('Test', 'https://myserver/opennms/', auth);
    * ```
-   * @example
-   * <caption>provide a username and password for auth</caption>
-   * ```javascript
-   * const server = new OnmsServer('Test', 'https://myserver/opennms/', 'admin', 'admin');
-   * ```
+   *
    * @constructor
    * @param name - A name for the server suitable for display.
    * @param url - The URL to the server.
    * @param auth - An [[OnmsAuthConfig]], or the username to authorize as.
    * @param password - The password to authorize with if a username was
    *                   supplied to the `auth` parameter.
    */
-  constructor(name?: string, url?: string, auth?: OnmsAuthConfig | string, password?: string) {
-    this.id = UUID.generate();
+  public constructor(name: string | undefined, url: string, auth?: OnmsAuthConfig, metadata?: ServerMetadata) {
```

Comment:
Same comment re: taking a builder as the argument

Reply:
resolving this comment and leaving the other one open for feedback to my feedback